ql/src/test/queries/clientpositive/prepare_plan.q

```diff
@@ -0,0 +1,49 @@
+--! qt:dataset:src
+--! qt:dataset:alltypesorc
+
+set hive.explain.user=false;
+--explain select * from src where key > '0' limit 10;
+set hive.vectorized.execution.enabled=false;
+
+
+explain extended prepare pcount as select count(*) from src where key > $1;
+prepare pcount as select count(*) from src where key > $1;
+execute pcount (1,200);
+select count(*) from src where key > '200';
+
+-- single param
+explain extended prepare p1 as select * from src where key > $1 order by key limit 10;
+explain extended select * from src where key > '200' order by key limit 10;
+prepare p1 as select * from src where key > $1 order by key limit 10;
+
+execute p1 (1,200);
+select * from src where key > '200' order by key limit 10;
+
+-- same query, different param
+execute p1 (1,0);
+select * from src where key > '0' order by key limit 10;
+
+-- multiple parameters
+explain prepare p2 as select min(ctinyint), max(cbigint) from alltypesorc where cint > ($1 + $2 + $3) group by ctinyint;
+prepare p2 as select min(ctinyint), max(cbigint) from alltypesorc where cint > ($1 + $2 + $3) group by ctinyint;
+
+execute p2 ( 1,0, 2,1, 3, 2);
+select min(ctinyint), max(cbigint) from alltypesorc where cint > (0 + 1 + 2) group by ctinyint;
```

Comment:
Since we are executing the query and verifying the results we may want to add an ORDER BY clause to guarantee that the order is not gonna change.